src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/GlobalSingleInstanceActivationMaintainer.cs

```diff
@@ -0,0 +1,334 @@
+                               ResponseException = e
+                           }, addresses.Count).ToArray());
+                }
+
+            }).ToList();
+ 
+            // wait for all the responses to arrive or fail
+            await Task.WhenAll(tasks);
+
+            if (logger.IsVerbose)
+            { 
+                foreach (var br in batchResponses)
+                {
+                    var summary = br.Aggregate(new { Pass = 0, Failed = 0, FailedA = 0, FailedOwned = 0, Faulted = 0 }, (agg, r) =>
+                    {
+                        switch (r.ResponseStatus)
+                        {
+                            case ActivationResponseStatus.Pass:
+                                return new { Pass = agg.Pass + 1, agg.Failed, agg.FailedA, agg.FailedOwned, agg.Faulted };
+                            case ActivationResponseStatus.Failed:
+                                if (!r.Owned)
+                                {
+                                    return r.ExistingActivationAddress.Address == null
+                                        ? new { agg.Pass, Failed = agg.Failed + 1, agg.FailedA, agg.FailedOwned, agg.Faulted }
+                                        : new { agg.Pass, agg.Failed, FailedA = agg.FailedA + 1, agg.FailedOwned, agg.Faulted };
+                                }
+                                else
+                                {
+                                    return new { agg.Pass, agg.Failed, agg.FailedA, FailedOwned = agg.FailedOwned + 1, agg.Faulted };
+                                }
+                            default:
+                                return new { agg.Pass, agg.Failed, agg.FailedA, agg.FailedOwned, Faulted = agg.Faulted + 1 };
+                        }
+                    });
+                    logger.Verbose("GSIP:M batchresponse PASS:{0} FAILED:{1} FAILED(a){2}: FAILED(o){3}: FAULTED:{4}",
+                        summary.Pass,
+                        summary.Failed,
+                        summary.FailedA,
+                        summary.FailedOwned,
+                        summary.Faulted);
+                }
+            }
+
+
+            // process each address
+
+            var loser_activations_per_silo = new Dictionary<SiloAddress, List<ActivationAddress>>();
+
+            for (int i = 0; i < addresses.Count; i++)
+            {
+                var address = addresses[i];
+
+                // array that holds the responses
+                var responses = new RemoteClusterActivationResponse[remoteClusters.Count];
+
+                for (int j = 0; j < batchResponses.Count; j++)
+                    responses[j] = batchResponses[j][i];
+
+                // response processor
+                var tracker = new GlobalSingleInstanceResponseTracker(responses, address.Grain);
+
+                tracker.CheckIfDone();
+
+                Debug.Assert(tracker.Task.IsCompleted);
+
+                var outcome = tracker.Task.Result;
```

Comment:
why is this creating a tracker if it assumes there is nothing to track and the response should be completed? Having a blocking .Result call here without making sure makes me a little nervous, as the code in the tracker might change and have this not be completed